checker/checker.go

```diff
@@ -144,6 +146,27 @@ func (v *visitor) BinaryNode(node *ast.BinaryNode) reflect.Type {
 	l := v.visit(node.Left)
 	r := v.visit(node.Right)
 
+	// check operator overloading
+	if fns, ok := v.operators[node.Operator]; ok {
+		for _, fn := range fns {
+			fnType, ok := v.types[fn]
+			if !ok || fnType.Type.Kind() != reflect.Func {
+				continue
```

Comment:
What if you specify operator overload on not existing helper's func? Here will be no errors. 
Maybe it's better to check overloads before starting compilation?